nginx/lua/gossip.lua

```diff
@@ -0,0 +1,132 @@
+local ngx = require("ngx")
+local cjson = require("cjson")
+local config = require("config")
+
+local Gossip = {
+}
+
+---@class PeerData
+---@field epoch integer Monotonic increasing number that is incremented by the peer
+---@field status string Current server status (TODO: enum?)
+---@field timestamp integer UNIX time of the last update
+---@field server_version string Version of the server
+
+
+---@type function
+---@return table<string, boolean> peers, boolean starting
+---Return a list of the current peers, and a boolean
+---indicating if this is the first time this worker is started
+function Gossip.peers()
+    local peerstr = ngx.shared.gossip_data:get("peers")
+    local starting = false
+    if not peerstr then
+        peerstr = config.data.seed_peers
+        ngx.shared.gossip_data:set("peers", peerstr)
+        starting = true
+    end
+    local peers = {}
+    ---@cast peerstr string
+    for peer in string.gmatch(peerstr, "[^,]+") do
+        peers[peer] = true
+    end
+    return peers, starting
+end
+
+---@type function
+---@param peer string
+---@return PeerData peerdata
+function Gossip.get_peerdata(peer)
```

Comment:
In the future we may want to start using cjson to ser/de the content rather than adding new dict entries for each new field